File: StudentRecord.cpp
#include "StudentRecord.h"


///////////////////////////
// I provide these functions
///////////////////////////
StudentRecord::StudentRecord(){}

std::string StudentRecord::firstname() const { return firstname_; }

std::string StudentRecord::lastname() const { return lastname_; }

double StudentRecord::score() const { return score_; }

std::ostream & operator<<( std::ostream & out, StudentRecord const &right ){ right.print(out); return out; }

std::istream & operator>>( std::istream & in, StudentRecord  &right ){ right.input(in); return in; }


///////////////////////////
// You provide these functions
///////////////////////////

bool StudentRecord::operator>( StudentRecord const & right ) const {
  return true; /// YOUR CODE GOES HERE
};

bool StudentRecord::operator>=( StudentRecord const & right ) const {
  return true; /// YOUR CODE GOES HERE
};

bool StudentRecord::operator<( StudentRecord const & right ) const {
  return true; /// YOUR CODE GOES HERE
};

bool StudentRecord::operator<=( StudentRecord const & right ) const {
  return true; /// YOUR CODE GOES HERE
};

void StudentRecord::compute_score(void) {
  /// YOUR CODE GOES HERE
}

void StudentRecordPhysics::print( std::ostream & out ) const  {
  /// YOUR CODE GOES HERE
}

bool StudentRecordPhysics::input( std::istream & in )  {
  /// YOUR CODE GOES HERE
  return true; 
}

void StudentRecordLiterature::print( std::ostream & out ) const  {
  /// YOUR CODE GOES HERE
}

bool StudentRecordLiterature::input( std::istream & in )  {
  /// YOUR CODE GOES HERE
  return true;  
}

void StudentRecordHistory::print( std::ostream & out ) const  {
  /// YOUR CODE GOES HERE
}

bool StudentRecordHistory::input( std::istream & in )  {
  /// YOUR CODE GOES HERE
  return true;
}

File: StudentRecord.py
import numpy as np

class StudentRecord :
    def __init__(self, line ) :
        self.firstname_ = ''
        self.lastname_ = ''
        self.score_ = None
        self.scores_ = None
        self.inputline(line)        
        
    def __str__(self):
        return self.lastname_ + ', ' + self.firstname_ + ' : ' + ' '.join([str(x) for x in self.scores_])

    def firstname(self):
        return self.firstname_

    def lastname(self):
        return self.lastname_

    def score(self):
        return self.score_

    def inputline(self,data):        
        self.lastname_ = data[0]
        self.firstname_ = data[1]
        self.scores_ = np.array([ float(x) for x in data[2:] ])

    def calcscore(self):
        ### your code goes here

class StudentRecordPhysics(StudentRecord):
    def __init__(self, line):
        ### your code goes here

class StudentRecordLiterature(StudentRecord):
    def __init__(self, line):
        ### your code goes here

class StudentRecordHistory(StudentRecord):
    def __init__(self, line):
        ### your code goes here

